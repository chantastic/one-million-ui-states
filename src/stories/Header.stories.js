import { Header } from "./Header";
import { fn } from "@storybook/test";

export default {
  title: "Example/Header",
  component: Header,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  args: {
    onLogin: fn(),
    onLogout: fn(),
    onCreateAccount: fn(),
  },
};

export const LoggedIn = {
  args: {
    user: {
      name: "Jane Doe",
    },
  },
};

export const LoggedOut = {};

/*
Lesson 3: Document extreme cases

Stories are isolated UI states.
They're cheap to create, making it easy to capture extreme cases.

Uncomment (or re-write) the code below to document both short and long extreme cases.
*/

export const LoggedInWithShortName = {
  args: {
    user: {
      name: "Jo",
    },
  },
};

export const LoggedInWithLongName = {
  args: {
    user: {
      name: "John Jacob Jingleheimer Schmidt",
    },
  },
};

/*
Lesson 4: Set story viewport

Using paramaters (in addition to args), we describe full UI cases.
Like the combination of a long name in small viewport.

Uncomment (or rewrite) the story below to see how long names render on mobile.
Note that this requires args from the `LoggedInWithLongName` story.
*/

export const LoggedInWithLongNameOnMobile = {
  args: LoggedInWithLongName.args,
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  },
};

/*
NOTE:
The viewport API is incredibly strange.
"mobile1" is a magic string in Storybook.
Read more about viewports here: https://storybook.js.org/docs/essentials/viewport#minimal_viewports
*/
