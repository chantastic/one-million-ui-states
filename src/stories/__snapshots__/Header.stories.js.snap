// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/Header LoggedIn smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome, ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Jane Doe",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-in",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Header LoggedInWithLongName smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome, ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "John Jacob Jingleheimer Schmidt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-in-with-long-name",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Header LoggedInWithLongNameOnMobile smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome, ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "John Jacob Jingleheimer Schmidt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-in-with-long-name-on-mobile",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Header LoggedInWithShortName smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome, ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Jo",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-in-with-short-name",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Header LoggedOut smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log in",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Sign up",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "example-header--logged-out",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
